src/open_inwoner/openklant/services.py

```diff
@@ -1096,19 +1105,24 @@ def get_or_create_digitaal_adres(
         partij_uuid: str,
         soortAdres: Literal["email", "telefoonnummer"],
         adres: str,
+        isStandaardAdres: bool = False,
     ) -> tuple[DigitaalAdres, bool]:
         digitale_adressen = self.filter_digitale_addressen_for_partij(
             partij_uuid, soortDigitaalAdres=soortAdres
         )
         for digitaal_adres in digitale_adressen:
-            if digitaal_adres["adres"] == adres:
+            if (
+                digitaal_adres["adres"] == adres
+                and digitaal_adres["isStandaardAdres"] is isStandaardAdres
```

Comment:
Because we're not directly comparing to `True`/`False` I feel that `==` might be a bit more readable than `is`, but that's a nitpick.

```suggestion
                and digitaal_adres["isStandaardAdres"] == isStandaardAdres
```